src/AlgebraicGeometry/Schemes/AffineSchemeOpenSubscheme/Rings/Methods.jl

```diff
@@ -243,7 +243,7 @@ function restriction_map(
   # the terms accordingly, we derive the desired expressions for the cᵢ's.
   #W = localized_ring(OO(Y))
   W = OO(Y)
-  S, t = polynomial_ring(W, ["t$i" for i in 1:r])
+  S, t = polynomial_ring(W, ["t$i" for i in 1:r]; cached=false)
```

Comment:
This is the only important change in this PR. The rest is just rewriting to make things more terse but otherwise has no semantic effect whatsoever and could also be dropped from this PR.